app/upgrades/v5/upgrades.go

```diff
@@ -179,3 +188,23 @@ func MigrateContributorClaim(ctx sdk.Context, k *claimskeeper.Keeper) {
 	k.DeleteClaimsRecord(ctx, from)
 	k.SetClaimsRecord(ctx, to, cr)
 }
+
+// UpdateConsensusParams
+func UpdateConsensusParams(ctx sdk.Context, sk stakingkeeper.Keeper, pk paramskeeper.Keeper) {
+	cp := ctx.ConsensusParams()
+	if cp == nil {
+		return
+	}
+
+	subspace, found := pk.GetSubspace(baseapp.Paramspace)
+	if !found {
+		return
+	}
+
+	maxAgeNumBlocks := sdk.NewInt(int64(cp.Evidence.MaxAgeDuration)).QuoRaw(int64(AvgBlockTime))
```

Comment:
@fedekunze Why are you using the pre-upgrade `MaxAgeDuration`? Shouldn't this move after line 207?

Reply:
good catch